test/gtest/dropout_ocl_hip.cpp

```diff
@@ -0,0 +1,825 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#include <gtest/gtest.h>
+#include <miopen/miopen.h>
+#include <miopen/handle.hpp>
+#include <miopen/dropout.hpp>
+#include <miopen/float_equal.hpp>
+#include <miopen/check_numerics.hpp>
+#include <miopen/tensor.hpp>
+#include <miopen/datatype.hpp>
+
+#include <rocrand/rocrand_xorwow.h>
+
+#include <vector>
+#include <numeric>
+#include <algorithm>
+#include <iostream>
+
+#include "get_handle.hpp"
+#include "random.hpp"
+#include "driver.hpp"
+#include "dropout_util.hpp"
+
+template <typename T>
+inline void SquashPairedTensor(const std::vector<T> x_len,
```

Comment:
need to document this code so I can understand what's going on. Please add comments.

Reply:
This code was taken from src/ocl/dropooutocl.cpp. I just wanted my modifications to be contained within the gtest so I did not modify the original source to make the function visible in my gtest.